gnucash/import-export/aqb/dialog-ab-select-imexporter.c

```diff
@@ -0,0 +1,257 @@
+    g_signal_connect (imexd->dialog, "destroy",
+                      G_CALLBACK (gtk_widget_destroyed), &imexd->dialog);
+    imexd->imexporter_list =
+        GTK_LIST_STORE (gtk_builder_get_object (builder, "imexporter-list"));
+    imexd->profile_list =
+        GTK_LIST_STORE (gtk_builder_get_object (builder, "profile-list"));
+    imexd->select_imexporter =
+        GTK_WIDGET (gtk_builder_get_object (builder, "imexporter-sel"));
+    imexd->select_profile =
+        GTK_WIDGET (gtk_builder_get_object (builder, "profile-sel"));
+    imexd->ok_button =
+        GTK_WIDGET (gtk_builder_get_object (builder, "imex-okbutton"));
+
+    imex_select = GTK_TREE_SELECTION (gtk_builder_get_object (builder, "imex-selection"));
+    prof_select = GTK_TREE_SELECTION (gtk_builder_get_object (builder, "prof-selection"));
+    populate_tree_model (GTK_TREE_MODEL (imexd->imexporter_list),
+                         imexporters);
+
+    g_signal_connect (imex_select, "changed", G_CALLBACK(imexporter_changed),
+                      imexd);
+    g_signal_connect (prof_select, "changed", G_CALLBACK(profile_changed),
+                      imexd);
+    g_list_free (imexporters);
+    g_object_unref (G_OBJECT (builder));
+
+    gtk_window_set_transient_for (GTK_WINDOW (imexd->dialog),
+                                  GTK_WINDOW (imexd->parent));
+
+    return imexd;
+}
+
+void
+gnc_ab_select_imex_dlg_destroy (GncABSelectImExDlg* imexd)
+{
+
+    if (imexd->imexporter_list)
+        clear_tree_model (GTK_TREE_MODEL (imexd->imexporter_list));
+
+    if (imexd->profile_list)
+        clear_tree_model (GTK_TREE_MODEL (imexd->profile_list));
+
+    if (imexd->dialog)
+        gtk_widget_destroy (imexd->dialog);
+
+    g_free (imexd);
+}
+
+gboolean
+imexporter_changed(GtkTreeSelection* sel, gpointer data)
+{
+    GncABSelectImExDlg* imexd = (GncABSelectImExDlg*)data;
+    GtkTreeIter iter;
+    GtkTreeModel* model;
+
+    gtk_widget_set_sensitive (imexd->ok_button, FALSE);
+
+    if (gtk_tree_selection_get_selected (sel, &model, &iter))
+    {
+        char* name = NULL;
+        GList* profiles = NULL;
+
+        gtk_tree_model_get (model, &iter, 0, &name, -1);
+        if (name && *name)
+        {
+             profiles = gnc_ab_imexporter_profile_list (imexd->abi, name);
+             g_free (name);
+        }
+        clear_tree_model (GTK_TREE_MODEL (imexd->profile_list));
+        if (profiles)
+        {
+             populate_tree_model (GTK_TREE_MODEL (imexd->profile_list), profiles);
+        }
+        else
+        {
+            gtk_widget_set_sensitive (imexd->ok_button, TRUE);
+            return FALSE;
+        }
+
+        if (!profiles->next)
+        {
+            GtkTreePath* path = gtk_tree_path_new_first();
+            GtkTreeSelection* profile_sel =
+                gtk_tree_view_get_selection (GTK_TREE_VIEW (imexd->select_profile));
+            gtk_tree_selection_select_path (profile_sel, path); //should call profile_changed
```

Comment:
I think you need to free the path with gtk_tree_path_free

Reply:
Right, thanks.